File: frontend/src/app/core/services/configuracion.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ConfiguracionRequest, ConfiguracionResponse } from '../models/configuracion.model';

@Injectable({
  providedIn: 'root'
})
export class ConfiguracionService {
  private http = inject(HttpClient);
  private apiUrl = 'http://localhost:8081/api/configuracion';

  obtenerConfiguracion(): Observable<ConfiguracionResponse> {
    return this.http.get<ConfiguracionResponse>(this.apiUrl);
  }

  actualizarConfiguracion(config: ConfiguracionRequest): Observable<ConfiguracionResponse> {
    return this.http.put<ConfiguracionResponse>(this.apiUrl, config);
  }
}

File: backend/src/main/java/com/insteip/backend/service/interfaces/ConfiguracionService.java
package com.insteip.backend.service.interfaces;

import com.insteip.backend.dto.ConfiguracionRequest;
import com.insteip.backend.dto.ConfiguracionResponse;

public interface ConfiguracionService {
    ConfiguracionResponse obtenerConfiguracion();
    ConfiguracionResponse actualizarConfiguracion(ConfiguracionRequest request);
}
